controller/src/main/java/org/jboss/as/controller/access/AuthorizationResult.java

```diff
@@ -46,20 +52,75 @@
     private final Decision decision;
     private final ModelNode explanation;
 
+    /**
+     * Creates an authorization result with no explanation.
+     *
+     * @param decision  the authorization decision. Cannot be {@code null}
+     */
     public AuthorizationResult(Decision decision) {
         this(decision, NO_EXPLANATION);
     }
 
+    /**
+     * Creates an authorization result with an optional explanation.
+     *
+     * @param decision  the authorization decision. Cannot be {@code null}
+     * @param explanation the explanation for the decision. May be {@code null}
+     */
     public AuthorizationResult(Decision decision, ModelNode explanation) {
         this.decision = decision;
         this.explanation = explanation == null ? NO_EXPLANATION : explanation;
     }
 
+    /**
+     * Gets the authorization decision.
+     *
+     * @return the decision. Will not be {@code null}
+     */
     public Decision getDecision() {
         return decision;
     }
 
+    /**
+     * Gets the explanation for the authorization decision. Will be an undefined node if no explanation
+     * was passed to the constructor.
+     *
+     * @return the explanation, an immutable model node. Will not be {@code null}, but may be an undefined node
+     */
     public ModelNode getExplanation() {
         return explanation;
     }
+
+    /**
+     * Utility method to throw a standard failure if {@link #getDecision()} is
+     * {@link org.jboss.as.controller.access.AuthorizationResult.Decision#DENY}.
+     * <p>
+     * This variant extracts the target address from the {@code address} field in the {@code operation} param
+     * and then calls the {@link #failIfDenied(ModelNode, PathAddress) overloaded variant}.
```

Comment:
Minor note, but the `{@link #failIfDenied(ModelNode, PathAddress) overloaded variant}` should probably be `{@linkplain #failIfDenied(ModelNode, PathAddress) overloaded variant}`.